drivers/storage/ide/uniata/id_probe.cpp

```diff
@@ -1967,16 +1970,24 @@ UniataClaimLegacyPCIIDE(
     if (!resourceList) {
         KdPrint2((PRINT_PREFIX "!resourceList\n"));
         status = STATUS_INSUFFICIENT_RESOURCES;
+#ifdef __REACTOS__
+        goto del_do;
+#else
 del_do:
         IoDeleteDevice(BMList[i].PciIdeDevObj);
         BMList[i].PciIdeDevObj          = NULL;
         return status;
+#endif
     }
 
     RtlZeroMemory(
         resourceList,
         sizeof(CM_RESOURCE_LIST));
 
+#ifdef __REACTOS__
+    oldResList = resourceList;
+#endif
+
     // IoReportDetectedDevice() should be used for WDM OSes
 
     // TODO: check if resourceList is actually used inside HalAssignSlotResources()
```

Comment:
OK, I've looked through this more careful and here is the thing: `resourceList` is not actually used inside `HalAssignSlotResources`. So as we're touching this code already, let's just remove all that resource allocation (the one which is now `oldResList`) stuff as not needed.

I don't see it being used on my version of Windows as well.

Reply:
Thats correct, `resourceList` is not used inside `HalAssignSlotResources` but according to the [code comment](https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/storage/ide/uniata/id_probe.cpp;h=1ca49d248ecdfc0b2f3379849776cf2f8a6b3a85;hb=HEAD#l1984), in some cases HalAssignSlotResources fails on some HW so a resource list is pre-allocated.
Now I would love to remove it, I really would but that would require extensive testing on all the supported hardware (virtual or otherwise). Which can be taken up as a task/bug later.
So for now I have settled on cleaning up the leaking memory

Reply:
Indeed, what's actually "used" by HalAssignSlotResources, is the pointer to pointer to the resource list, so that it knows where to store the pointer to the new allocated list. So, strange, but all the initialization code for resourceList done just before HalAssignSlotResources in principle is thus "dead" code.

Reply:
Yep. Its mostly some other issue and not if the resource list is pre-allocated or not but I don't want to break the code